process_report/invoices/MOCA_group_specific_invoice.py

```diff
@@ -0,0 +1,201 @@
+        group_credit_info = self.prepay_credits[group_credit_mask]
+        if not group_credit_info.empty:
+            group_credit = group_credit_info.squeeze()[invoice.PREPAY_CREDIT_FIELD]
+            group_projects.loc[len(group_projects)] = None
+
+            existing_columns = self._get_existing_columns(
+                group_projects, self.CREDIT_COLUMN_COPY_LIST
+            )
+            for column_name in existing_columns:
+                group_projects.loc[
+                    group_projects.index[-1], column_name
+                ] = group_projects.loc[0, column_name]
+
+            group_projects.loc[
+                group_projects.index[-1], [invoice.COST_FIELD, invoice.BALANCE_FIELD]
+            ] = [group_credit] * 2
+
+        # Add sum row
+        group_projects.loc[len(group_projects)] = None
+        existing_columns = self._get_existing_columns(
+            group_projects, self.TOTAL_COLUMN_LIST
+        )
+        group_projects.loc[
+            group_projects.index[-1], invoice.INVOICE_DATE_FIELD
+        ] = "Total"
+        group_projects.loc[group_projects.index[-1], existing_columns] = group_projects[
+            existing_columns
+        ].sum()
+
+        # Add dollar signs
+        existing_columns = self._get_existing_columns(
+            group_projects, self.DOLLAR_COLUMN_LIST
+        )
+        for column_name in existing_columns:
+            group_projects[column_name] = group_projects[column_name].apply(
+                add_dollar_sign
+            )
+
+        group_projects.fillna("", inplace=True)
+
+        return group_projects
+
+    def export(self):
+        def _create_html_invoice(temp_fd):
+            environment = Environment(loader=FileSystemLoader(TEMPLATE_DIR_PATH))
+            template = environment.get_template("pi_invoice.html")
+            content = template.render(
+                data=group_dataframe,
+            )
+            temp_fd.write(content)
+            temp_fd.flush()
+
+        def _create_pdf_invoice(temp_fd_name):
+            chrome_binary_location = os.environ.get(
+                "CHROME_BIN_PATH", "usr/bin/chromium"
+            )
+            if not os.path.exists(chrome_binary_location):
+                sys.exit(
+                    f"Chrome binary does not exist at {chrome_binary_location}. Make sure the env var CHROME_BIN_PATH is set correctly or that Google Chrome is installed"
+                )
+
+            invoice_pdf_path = f"{self.name}/{group_instituition}_{group_contact_email}_{self.invoice_month}.pdf"
+            subprocess.run(
+                [
+                    chrome_binary_location,
+                    "--headless",
+                    "--no-sandbox",
+                    f"--print-to-pdf={invoice_pdf_path}",
+                    "--no-pdf-header-footer",
+                    "file://" + temp_fd_name,
+                ],
+                capture_output=True,
+            )
```

Comment:
This looks like a duplicate of the code in `process_report/invoices/pi_specific_invoice.py`. Can we factor this out into a common base class (or mixin)?

Reply:
Because I believe #137 will undergo a lot of changes during review I will factor this into a base class after #137 is merged.